backend/de.metas.manufacturing/src/main/java/org/eevolution/model/validator/PP_Order.java

```diff
@@ -275,4 +278,19 @@ public void validateBOMAndProduct(@NonNull final I_PP_Order ppOrder)
 					.setParameter("PP_Order.PP_Product_BOM_ID.M_Product_ID", productIdOfBOM.getRepoId());
 		}
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE })
+	public void setSeqNo(final I_PP_Order ppOrder)
+	{
+		if (ppOrder.getSeqNo() <= 0)
```

Comment:
i'm not sure about this new MI method:
1st concern: i think we need an index on SeqNo; otherwise metasfresh will needs to do a full sequential scan other all PP_Orders
2nd concern: why did you use this code, and not a `@SQL=...` expression? so far afair we used that expression whenever we can, so according to the principle of least surprise, i would expect it there as well.

@teosarca wdyt?

Reply:
@riadmen  @metas-ts I have a much more radical question/suggestion.
What's the purpose of PP_Order.SeqNo.... because I see none, just to confuse me as a user.

My suggestion would be to drop it (i.e. don't introduce PP_Order.SeqNo),
but just order by PP_Order_Candidate.SeqNo when generating PP_Orders (as you actually do in this task!).